src/main/java/org/opensearch/knn/indices/ModelDao.java

```diff
@@ -437,51 +464,143 @@ public void delete(String modelId, ActionListener<DeleteModelResponse> listener)
                 return;
             }
 
+            StepListener<GetModelResponse> getModelStep = new StepListener<>();
+            StepListener<AcknowledgedResponse> blockModelIdStep = new StepListener<>();
+            StepListener<AcknowledgedResponse> clearModelMetadataStep = new StepListener<>();
+            StepListener<DeleteResponse> deleteModelFromIndexStep = new StepListener<>();
+            StepListener<RemoveModelFromCacheResponse> clearModelFromCacheStep = new StepListener<>();
+            StepListener<AcknowledgedResponse> unblockModelIdStep = new StepListener<>();
+
+            // Get Model to check if model is in TRAINING
+            get(modelId, ActionListener.wrap(getModelStep::onResponse, exception -> {
+                if (exception instanceof ResourceNotFoundException) {
+                    String errorMessage = String.format("Unable to delete model \"%s\". Model does not exist", modelId);
+                    listener.onFailure(new ResourceNotFoundException(errorMessage));
+                    return;
+                }
+                listener.onFailure(exception);
+            }));
+
+            getModelStep.whenComplete(getModelResponse -> {
+                // If model is in Training state, fail delete model request
+                if (ModelState.TRAINING.equals(getModelResponse.getModel().getModelMetadata().getState())) {
```

Comment:
if both are enum members we can use "==" to compare, it's also null safe

```suggestion
                if (ModelState.TRAINING == getModelResponse.getModel().getModelMetadata().getState()) {
```

Reply:
ack